restapi/src/main/java/io/stargate/web/models/udt/UdtInfo.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.models.udt;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import io.swagger.annotations.ApiModelProperty;
+import java.util.List;
+import javax.validation.constraints.NotNull;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UdtInfo {
+  private String name;
+  private boolean frozen;
+  private @NotNull List<UdtType> typeParams;
+
+  @ApiModelProperty(value = "return the name")
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @ApiModelProperty(value = "return true if is frozen")
+  public boolean isFrozen() {
+    return frozen;
+  }
+
+  public void setFrozen(boolean frozen) {
+    this.frozen = frozen;
+  }
+
+  @JsonProperty("typeParams")
+  @ApiModelProperty(value = "return type parameters")
```

Comment:
```java
@ApiModelProperty(value = "The CQL type of the parameters being used (e.g. text, int, date)")
```

Reply:
Or something to that effect.

Reply:
I'm guessing this also needs `required = true`?

Reply:
I'm amending my original suggestion after reading further. So this here is actually the sub-type which we should definitely call out in the swagger. 

```java
@ApiModelProperty(value = "The CQL sub-type of the parent type. This is to be used when the parent type is a collection (list, tuple, map, or set) or a UDT.")
```

Reply:
> I'm guessing this also needs required = true?
IIRC, in the case of the Tuple it doesn't need a sub-type.